packages/http-client/src/middleware/middleware-type.ts

```diff
@@ -0,0 +1,77 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { HttpRequestConfig } from '../http-client-types';
+
+const logger = createLogger('middleware');
+
+/**
+ * In/out parameter in the chain of middlewares.
+ */
+export interface MiddlewareInOut<ReturnType, ContextType extends Context> {
+  /**
+   * Function execute inside the middleware.
+   */
+  fn: () => Promise<ReturnType>;
+  /**
+   * Context of the execution.
+   */
+  context: ContextType;
+  /**
+   * Exit the middleware chain.
+   */
+  exitChain: boolean;
+}
+
+/**
+ * Minimal Context of the middleware.
+ */
+export interface Context {
+  /**
+   * Arguments used in the request.
+   */
+  args: any[];
+  /**
+   * URI of the request wrapped in the middleware.
+   */
+  uri: string;
+}
+
+/**
+ * Middleware type.
+ */
+export type Middleware<ReturnType, ContextType extends Context> = (
```

Comment:
[pp] In other places we have used `EntityT` and `ReturnT`, maybe we should continue this here as well (=> `<ReturnT, ContextT>`?

Reply:
sure